daml/Marketplace/Trading/Order.daml

```diff
@@ -0,0 +1,83 @@
+module Marketplace.Trading.Order where
+
+import DA.Finance.Asset (AssetDeposit)
+import DA.Finance.Types (Id(..), Asset(..))
+
+data Side
+    = Buy
+    | Sell
+  deriving (Eq, Show)
+
+data OrderType
+    = Market
+    | Limit with
+        price : Decimal
+  deriving (Eq, Show)
+
+data TimeInForce
+    = GTC
+      -- ^ Good Till Cancelled (Rests on book until cancellation)
+    | GTD with
+        expiryDate : Int
+      -- ^ Good Till Date (At expiryDate, order will be automatically cancelled).
+      -- UTC date and time in seconds
+    | GAA
+      -- ^ Good At Auction (Expires after auction if not filled)
+    | IOC
+      -- ^ Immediate Or Cancel (Allows for partial fills)
+    | FOK
+      -- ^ Fill Or Kill (All or nothing)
+  deriving (Eq, Show)
+
+data OrderDetails = OrderDetails with
+    id : Id
+    symbol : Text
+    asset : Asset
+    side : Side
+    orderType : OrderType
+    timeInForce : TimeInForce
+  deriving (Eq, Show)
+
+data Status
+    = New
+    | PendingExecution
+    | PartiallyExecuted
+    | FullyExecuted
+    | Rejected with
+        errorCode : Int
+        errorMessage : Text
+    | PendingCancellation
+    | CancellationRejected with
+        errorCode : Int
+        errorMessage : Text
+    | Cancelled
+  deriving (Eq, Show)
+
+data ExecutionFill = ExecutionFill with
+    matchId : Text
+    makerOrderId : Text
+    takerOrderId : Text
+    executedQuantity : Decimal
+    executedPrice : Decimal
+    executedTimestamp : Text -- Should this be an Int so we can order if necessary?
+  deriving (Eq, Show)
+
+type T = Order
+
+template Order
+  with
+    operator : Party
+    provider : Party
+    customer : Party
+    status : Status
+    orderDetails : OrderDetails
+    providerOrderId : Optional Text
```

Comment:
`ExchangeOrderId` to avoid ambiguity (provider could be the one who originated the order)

Reply:
This was `ExchangeOrderId` but Georg renamed this to `provider` so I presume to abstract that we may or may not be using an actual Exchange underneath.